src/server/server_family.cc

```diff
@@ -903,11 +920,16 @@ GenericError ServerFamily::DoSave(bool new_version, Transaction* trans) {
     service_.SwitchState(GlobalState::SAVING, GlobalState::ACTIVE);
   };
 
-  const auto& dbfilename = GetFlag(FLAGS_dbfilename);
-  fs::path filename = dbfilename.empty() ? "dump" : dbfilename;
+  absl::Time start = absl::Now();
+
+  fs::path filename = GetFlag(FLAGS_dbfilename);
+  if (!ValidateFilenameExtension(filename, new_version)) {
```

Comment:
make ValidateFilenameExtension return generic error code and have the "Bad filename extension ", filename.extension().c_str() inside it